hadoop-ozone/dist/src/main/k8s/examples/testlib.sh

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env bash
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+retry() {
+   n=0
+   until [ $n -ge 30 ]
+   do
+      "$@" && break
+      n=$[$n+1]
+      echo "$n '$@' is failed..."
+      sleep ${RETRY_SLEEP:-3}
+   done
+   if [ $n -eq 30 ]; then
+      return 255
+   fi
+}
+
+grep_log() {
+   CONTAINER="$1"
+   PATTERN="$2"
+   kubectl logs "$1"  | grep "$PATTERN"
+}
+
+wait_for_startup(){
+   print_phase "Waiting until the k8s cluster is running"
+   retry all_pods_are_running
+   retry grep_log scm-0 "SCM exiting safe mode."
+   retry grep_log om-0 "HTTP server of ozoneManager listening"
+   print_phase "Cluster is up and running"
+}
+
+all_pods_are_running() {
+   RUNNING_COUNT=$(kubectl get pod --field-selector status.phase=Running | wc -l)
+   ALL_COUNT=$(kubectl get pod | wc -l)
+   RUNNING_COUNT=$((RUNNING_COUNT - 1))
+   ALL_COUNT=$((ALL_COUNT - 1))
+   if [ "$RUNNING_COUNT" -lt "3" ]; then
+      echo "$RUNNING_COUNT pods are running. Waiting for more."
+      return 1
+   elif [ "$RUNNING_COUNT" -ne "$ALL_COUNT" ]; then
+      echo "$RUNNING_COUNT pods are running out from the $ALL_COUNT"
+      return 2
+   else
+      STARTED=true
+      return 0
+   fi
+}
+
+start_k8s_env() {
+   print_phase "Deleting existing k8s resources"
+   #reset environment
+   kubectl delete pvc --all
+   kubectl delete pv --all
+   kubectl delete statefulset --all
+   kubectl delete daemonset --all
+   kubectl delete deployment --all
+   kubectl delete service --all
+   kubectl delete configmap --all
+   kubectl delete pod --all
+
+   print_phase "Applying k8s resources from $1"
+   kubectl apply -f .
+   wait_for_startup
+}
+
+stop_k8s_env() {
+   if [ ! "$KEEP_RUNNING" ]; then
+     kubectl delete -f .
+   fi
+}
+
+regenerate_resources() {
+  PARENT_OF_PARENT=$(realpath ../..)
+
+  if [ $(basename $PARENT_OF_PARENT) == "k8s" ]; then
+    #running from src dir
+    OZONE_ROOT=$(realpath ../../../../../target/ozone-0.6.0-SNAPSHOT)
+  else
+    #running from dist
+    OZONE_ROOT=$(realpath ../../..)
+  fi
+
+  flekszible generate -t mount:hostPath="$OZONE_ROOT",path=/opt/hadoop -t image:image=apache/ozone-runner:20200420-1 -t ozone/onenode
```

Comment:
Is there any way to avoid hard-coding the image name?

Reply:
It can be replaced easily during build time, but today it's possible to execute the tests from `src/min/k8s`, which makes it harder. I can parse it from the `pom.xml` but it's only for development.

I am not sure what is the most simple solution...

Reply:
OK, I'm fine with including it here.  Maybe a note near the version property in `pom.xml` would be nice to avoid someone forgetting to update this.